tests/test_features.py

```diff
@@ -10,14 +15,6 @@
 except KeyError:
     pass
 
-import matplotlib
```

Comment:
With that in mind you can also see what the "transposition error" is: You're not iterating along `axis` but, in the 2D case, along the other one. `axis` is then the operation direction inside the iteration loop.

That's also the reason why [the documented behaviour](https://github.com/bmcfee/librosa/blob/master/librosa/util/utils.py#L594) is so unexpected. NumPy usually does it the other way round (`axis=0` are rows). In the 2D case you simply swap an `axis=0` with an `axis=1` and waste no further thought on it.

In the 3D case however it really messes with your understanding of what is happening :-)

Changing this however will be a compatibility breaking change...